processing/src/main/java/org/apache/carbondata/processing/sortandgroupby/sortdata/SortDataRows.java

```diff
@@ -193,20 +193,42 @@ public void startSorting() throws CarbonSortKeyAndGroupByException {
       toSort = new Object[entryCount][];
       System.arraycopy(recordHolderList, 0, toSort, 0, entryCount);
 
-      if (parameters.isUseKettle()) {
-        if (parameters.getNoDictionaryCount() > 0) {
-          Arrays.sort(toSort, new RowComparator(parameters.getNoDictionaryDimnesionColumn(),
-              parameters.getNoDictionaryCount()));
+      if (parameters.getNumberOfSortColumns() == 0) {
```

Comment:
I feel, we should not use the old flow.  If user don't specify sort_columns then we should add all dimensions to sort_columns. In this way in our core/processing always depends upon sort_columns, so that code would be clear.  We just always depends on the sort_columns in our core code, please refactor the code in that way.